app/features/syllabus_generator/tools.py

```diff
@@ -0,0 +1,253 @@
+import os
+from typing import Dict, List
+
+from app.features.syllabus_generator.schema import SyllabusModel
+from app.services.logger import setup_logger
+from langchain_core.output_parsers import JsonOutputParser
+from langchain_core.prompts import PromptTemplate
+from langchain_google_genai import GoogleGenerativeAI
+from pydantic import ValidationError
+
+logger = setup_logger(__name__)
+
+
+def read_text_file(file_path: str) -> str:
+    """Read the content of a text file in the syllabus_generator dir."""
+    script_dir = os.path.dirname(os.path.abspath(__file__))
+    absolute_file_path = os.path.join(script_dir, file_path)
+    with open(absolute_file_path, "r") as file:
+        return file.read()
+
+
+ALLOWED_OPTIONS = {
+    "all": None,
+    "title": "Include a title for the syllabus.",
+    "overview": "Generate a comprehensive overview of the course.",
+    "objectives": "Generate a list of learning objectives to show a student's understanding.",
+    "required_materials": "Generate a list of materials that are required by the students to successfully participate in the course. Materials could be things like books, tools or supplies. Use two key variables in the output of required_materials. One is recommended_books and another is required_items.",
+    "additional_information": "Include any additional information that could be useful to the student",
+    "policies_and_exceptions": "Create a dictionary of policies and exceptions that apply to the course.",
+    "grade_level_assessments": "{grade_level_assessments}",
+}
+
+# Options that can be added to prompt
+USABLE_OPTIONS = ALLOWED_OPTIONS.copy()
+USABLE_OPTIONS.pop("all", None)
+
+
+class SyllabusBuilder:
+    def __init__(
+        self,
+        subject: str,
+        grade_level: str,
+        course_overview: str = "",
+        customisation: str = "",
+        options: List[str] = ["all"],
+        prompt: str = "",
+        model=None,
+        parser=None,
+        verbose: bool = False,
+    ):
+        """Initialize SyllabusBuilder with default configurations."""
+        default_config = {
+            "model": GoogleGenerativeAI(model="gemini-1.0-pro"),
```

Comment:
Replace this model to `gemini-1.5-flash` or `gemini-1.5-pro` as that model is deprecated.